shap/plots/_beeswarm.py

```diff
@@ -880,11 +885,31 @@ def color(i):
             class_inds = np.argsort([-np.abs(shap_values[i]).mean() for i in range(len(shap_values))])
         elif class_inds == "original":
             class_inds = range(len(shap_values))
+
+        if show_values_in_legend:
+            # Get the smallest decimal place of the first significant digit
+            # to print on the legend. The legend will print ('n_decimal' + 1)
+            # decimal places.
+            # Set to 1 if the smallest number is bigger than 1.
+            smallest_shap = np.abs(np.min(np.abs(shap_values).mean((1, 2))))
+            if smallest_shap > 1:
+                n_decimals = 1
+            else:
+                n_decimals = int(-np.floor(
+                    np.log10(
+                        smallest_shap
+                    )
+                ))
+
         for i, ind in enumerate(class_inds):
             global_shap_values = np.abs(shap_values[ind]).mean(0)
+            if show_values_in_legend:
+                label = f'{class_names[ind]}: {np.round(np.mean(global_shap_values),(n_decimals+1))}'
+            else:
+                label = class_names[ind]
```

Comment:
In terms of the formatting of the legend, I just want to note here that the cohort bar plots are already using `cohort_name [cohort_size]`

![image](https://github.com/slundberg/shap/assets/30731072/f6346f8d-8627-4c3c-9c31-5b4b1b023867)

---

And in `decision_plot` we have `group_name (shap_value)` kind of label formatting.

![image](https://github.com/slundberg/shap/assets/30731072/ee7c6fa4-5ca1-46b4-bf79-28a26876c9c5)

Can we change the label formatting to use parenthesis so we conform more to the same "look" as decision plot?

We might have to relook into adjusting how the labels look in the future, when we eventually add multi-output support to `shap.plots.bar`, though.